src/alert/alert.ts

```diff
@@ -85,7 +92,12 @@ export class NgbDismissibleAlert implements OnInit {
       this.closeEvent.emit($event);
       this.close();
     });
+    if (this.dismissOnTimeout) {
+      this._timeout = setTimeout(() => { this.close(); }, toInteger(this.dismissOnTimeout));
```

Comment:
I'm surprised that it works in a constructor... Did you test it in plunker to see if it behaves properly in the "real world"? So far I was living with an assumption that inputs are not defined in a constructor (change detection didn't run yet). 

Hmm, actually it _might_ work for literals but not when we use a variable. We need to do more testing / exploration here and eventually move the timeout-setting logic to `ngOnInit`